services/ui-src/src/hooks/authHooks/requestOptions.js

```diff
@@ -1,12 +1,10 @@
-import { Auth } from "aws-amplify";
+import { fetchAuthSession } from "aws-amplify/auth";
 
 async function requestOptions(body = null) {
   try {
-    const session = await Auth.currentSession();
-    const token = await session.getIdToken().getJwtToken();
-
+    const { idToken } = (await fetchAuthSession()).tokens ?? {};
```

Comment:
Thought: I'm not a fan of Amplify's typing here. For reference, `fetchAuthSession()` returns
```ts
Promise<{
  tokens?: {
    idToken?: JWT;
    //...
  };
  //...
}>
```
And it feels... catty. "maybe I'll give you tokens, and maaaybe one of them will be an ID token". Which puts us in a position where we have to check for the truthiness of `tokens` and `idToken`. If either is missing, we'll get a runtime error below ("cannot read properties of undefined (reading 'toString')"). To be clear, this is on AWS and not on you.

I think I might prefer an explicit error here over defaulting to undefined, something like
```suggestion
    const apiKey = (await fetchAuthSession()).tokens?.idToken?.toString();
    if (apiKey === undefined) {
      throw new Error("Amplify has betrayed us");
    }
```

But maybe that's going overboard / being too paranoid. If Amplify is being overly... evasive about its types, and the token & idToken are always present, then we will have just added noise to the code, accomplishing nothing

Reply:
The `?? {}` is their own recommendation for handling it. The whole body.json() thing has a similar vibe. I think I can add a more explicit error here.

Reply:
Up to you!